code/modules/admin/callproc.dm

```diff
@@ -0,0 +1,297 @@
+		log_admin(msg)
+		message_admins(msg) //Proccall announce removed.
+		admin_ticket_log(target, msg)
+		returnval = WrapAdminProcCall(target, procname, lst) // Pass the lst as an argument list to the proc
+	else
+		//this currently has no hascall protection. wasn't able to get it working.
+		log_admin("[key_name(src)] called [procname]() with [lst.len ? "the arguments [list2params(lst)]":"no arguments"].")
+		message_admins("[key_name(src)] called [procname]() with [lst.len ? "the arguments [list2params(lst)]":"no arguments"].") //Proccall announce removed.
+		returnval = WrapAdminProcCall(GLOBAL_PROC, procname, lst) // Pass the lst as an argument list to the proc
+	if(get_retval)
+		get_retval += returnval
+	. = get_callproc_returnval(returnval, procname)
+	if(.)
+		to_chat(usr, ., confidential = TRUE)
+
+/// Wrapper for proccalls where the datum is flagged as vareditted
+/proc/WrapAdminProcCall(datum/target, procname, list/arguments)
+	if(target && procname == "Del")
+		to_chat(usr, "Calling Del() is not allowed", confidential = TRUE)
+		return
+
+	if(target != GLOBAL_PROC && !target.CanProcCall(procname))
+		to_chat(usr, "Proccall on [target.type]/proc/[procname] is disallowed!", confidential = TRUE)
+		return
+
+	var/current_caller = GLOB.AdminProcCaller
+	var/user_identifier = usr ? usr.client?.ckey : GLOB.AdminProcCaller
+	var/is_remote_handler = usr == GLOB.AdminProcCallHandler
+	if(is_remote_handler)
+		user_identifier = GLOB.AdminProcCallHandler.name
+
+	if(!user_identifier)
+		CRASH("WrapAdminProcCall with no ckey: [target] [procname] [english_list(arguments)]")
+
+	if(!is_remote_handler && current_caller && current_caller != user_identifier)
+		to_chat(usr, SPAN_ADMINNOTICE("Another set of admin called procs are still running. Try again later."), confidential = TRUE)
+		return
+
+	GLOB.LastAdminCalledProc = procname
+	if(target != GLOBAL_PROC)
+		GLOB.LastAdminCalledTargetRef = REF(target)
+
+	if(!is_remote_handler)
+		GLOB.AdminProcCaller = user_identifier //if this runtimes, too bad for you
+		GLOB.AdminProcCallCount++
+		. = world.WrapAdminProcCall(target, procname, arguments)
+		if(GLOB.AdminProcCallCount-- == 0)
+			GLOB.AdminProcCaller = null
+	else
+		. = world.WrapAdminProcCall(target, procname, arguments)
+
+//adv proc call this, ya nerds
+/world/proc/WrapAdminProcCall(datum/target, procname, list/arguments)
+	if(target == GLOBAL_PROC)
+		return call("/proc/[procname]")(arglist(arguments))
+	else if(target != world)
+		return call(target, procname)(arglist(arguments))
+	else
+		log_admin("[key_name(usr)] attempted to call world/proc/[procname] with arguments: [english_list(arguments)]")
+
+/proc/IsAdminAdvancedProcCall()
+	return (GLOB.AdminProcCaller && GLOB.AdminProcCaller == usr?.client?.ckey) || (GLOB.AdminProcCallHandler && usr == GLOB.AdminProcCallHandler)
+
+/client/proc/callproc_datum(datum/called_datum)
```

Comment:
```suggestion
/client/proc/callproc_datum(datum/called_datum as null|area|mob|obj|turf)
```
For whatever reason, Datum ProcCall does not appear on right click menu the way Atom ProcCall did. I know nothing about underlying Byond sorceries here, but any chance this difference between definitions might be responsible?